src/Npgsql/Replication/Logical/Protocol/UpdateMessage.cs

```diff
@@ -0,0 +1,35 @@
+﻿using JetBrains.Annotations;
+using NpgsqlTypes;
+using System;
+
+namespace Npgsql.Replication.Logical.Protocol
+{
+    /// <summary>
+    /// Logical Replication Protocol update message for tables with REPLICA IDENTITY set to DEFAULT.
+    /// </summary>
+    /// <remarks>
+    /// This is the base type of all update messages containing only the tuples for the new row.
+    /// </remarks>
+    [PublicAPI]
+    public class UpdateMessage : LogicalReplicationProtocolMessage
+    {
+        internal UpdateMessage(NpgsqlLogSequenceNumber walStart, NpgsqlLogSequenceNumber walEnd, DateTime serverClock,
+            uint relationId, ITupleData[] newRow) : base(walStart, walEnd, serverClock)
+        {
+            RelationId = relationId;
+            NewRow = newRow;
+        }
+
+        /// <summary>
+        /// ID of the relation corresponding to the ID in the relation message.
+        /// </summary>
+        [PublicAPI]
+        public uint RelationId { get; }
+
+        /// <summary>
+        /// Columns representing the new row.
+        /// </summary>
+        [PublicAPI]
+        public ITupleData[]  NewRow { get; }
```

Comment:
👍 
I initially had a solution with a self-implemented `IReadonlyCollection` to avoid mutability but then I decided I don't want to loose the efficiency of arrays and add another boxing because of this.
I didn't think about `ReadOnlyMemory<ITupleData>` which obviously is the solution here.